### PR TITLE
[scheduler] Halve WakerPage struct size

### LINKED ISSUE #175: [scheduler] Remove drop bits
Description
========

We currently have 32 bytes in the `WakerPage` unused. We should make something useful out of this space.

One simple idea would be to halve the size of the structure itself.

### PR DESCRIPTION
**Pull request by anandbonde**
Related to #175

### CODE REVIEW

File: src/rust/scheduler/waker64.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ use ::std::{
 //==============================================================================
 
 /// Log2 of [WAKER_BIT_LENGTH].
-pub const WAKER_BIT_LENGTH_SHIFT: usize = 6;
+pub const WAKER_BIT_LENGTH_SHIFT: usize = 5;
 
 /// Number of Bits in a [Waker64]
 pub const WAKER_BIT_LENGTH: usize = 1 << WAKER_BIT_LENGTH_SHIFT;
```
Comment (ppenna) [inline src/rust/scheduler/waker64.rs:21]: This changes more than it appears.

**Review by ppenna — CHANGES_REQUESTED** [review]
It looks like you were attempting to halve the structure entirely, as it is suggested in the issue description. Unfortunately, this would require major rework, because sizes of `WakerPage` and `Waker64` are very coupled. The correct fix int hat direction would be change everything to be 32-bit long. Note however, that this would have a major negative impact on 64 bit-machine, when traversing the scheduler bitmap. Instead, I would suggest us to really use that unsued field for something more interesting, such as introducing a priority-based scheduling with 2 scheduling levels (zero's and ones).

For instance, the way that we traverse

File: src/rust/scheduler/waker64.rs (hunk 1 above)
Comment (iyzhang) [reply src/rust/scheduler/waker64.rs:21]: Lol, this is the most mysterious comment that I have ever seen.

Comment (ppenna) [reply src/rust/scheduler/waker64.rs:21]: I just wanted to point out that this is the place where we actually change the length of `WAKER`s.

**Comment by iyzhang** [comment]
This issue is related to a recent one that I posted #723

**Comment by anandbonde** [comment]
Closing this PR, @iyzhang will handle this in a separate PR.

**Pull request closed without merging**